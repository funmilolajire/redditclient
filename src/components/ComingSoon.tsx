import { useRef, useLayoutEffect } from 'react';
import { gsap } from 'gsap';

export const ComingSoon = () => {
    const boxRef = useRef<HTMLDivElement>(null);

    useLayoutEffect(() => {
        const tl = gsap.timeline({ repeatDelay: 2 })
        tl.from([boxRef.current], {
            y: '-55vh',
            duration: 3,
            delay: 2,
            opacity: 0,
            ease: 'elastic'
        })
    })

    return (
        <div ref={boxRef} className="container flex items-center justify-center h-screen font-extrabold ComingSoon">
            <div className="bg-yellow-500 border-2 border-gray-800 border-solid skew-y-2 transform-gpu text-gray-50 text-9xl">
                <p>Coming Soon</p>
            </div>
        </div>
    )
}