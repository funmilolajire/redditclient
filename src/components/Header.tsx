import { useRef, useLayoutEffect } from 'react';
import { gsap } from 'gsap';
import logo from '../images/logo192.png';

export const Header = () => {
    const boxRef = useRef<HTMLDivElement>(null);
    const logoRef = useRef<HTMLImageElement>(null);
    const titleRef = useRef<HTMLHeadingElement>(null);

    useLayoutEffect(() => {
        const tl = gsap.timeline({});
        tl.from([boxRef.current], {
            duration: 1
        })
        tl.from([logoRef.current], {
            x: '-3000px',
            opacity: 0,
            ease: 'circ'
        })
        tl.from([titleRef.current], {
            x: '3000px',
            opacity: 0,
            ease: 'slow'
        })
    })

    return (
        <header>
            <div className="header-inner">
                <section ref={boxRef}>
                    <figure ref={logoRef}>
                        <img src={logo} alt="site logo" />
                    </figure>
                    <h1 ref={titleRef}>
                        <span>Reddit</span>Client
                    </h1>
                </section>
            </div>
        </header>
    )
}