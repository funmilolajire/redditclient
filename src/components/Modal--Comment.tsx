import { useRef } from 'react';
import parse from 'html-react-parser';
import { BiComment } from 'react-icons/bi';
import { BsDot } from 'react-icons/bs';
import { GoArrowDown, GoArrowUp } from 'react-icons/go';

interface ModalCommentType {
    [key: string]: any
}

export const ModalComment = ({ comment }: { comment: ModalCommentType }) => {
    const modalCommentRef = useRef<HTMLDivElement>(null);

    //parse comment html
    const parsedHTML = (html: string) => html ? parse(parse(html).toString()) : null;

    //JSX Element
    return (
        <div className="ModalComment" ref={modalCommentRef} data-permalink={comment.permalink}>
            <p className="Comment--Info">comment made <span className="time">{comment.time}</span> by <span className="author">{comment.author}</span></p>
            <div className="Comment--Body">
                {parsedHTML(comment.comment)}
            </div>
            <div className="Comment--Bottom">
                <div className="Comment--Replies">
                    <BiComment className="reply-icon icon" />
                    <span className="comments value">{comment.num_replies}</span>
                    <span className="label">{comment.num_replies !== 1 ? 'replies' : 'reply'}</span>
                </div>
                <BsDot className="dot-icon" />
                <div className="Votes Comment--Upvotes">
                    <GoArrowUp className="arrow-icon icon" />
                    <span className="upvotes value">{comment.upvotes}</span>
                    <span className="label">{comment.upvotes !== 1 ? 'upvotes' : 'upvote'}</span>
                </div>
                <BsDot className="dot-icon" />
                <div className="Votes Comment--Downvotes">
                    <GoArrowDown className="arrow-icon icon" />
                    <span className="downvotes value">{comment.downvotes}</span>
                    <span className="label">{comment.downvotes !== 1 ? 'downvotes' : 'downvote'}</span>
                </div>
            </div>
        </div>
    )
}