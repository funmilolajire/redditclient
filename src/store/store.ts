import { configureStore } from '@reduxjs/toolkit';
import termReducer from './slices/termSlice';

export default configureStore({
  reducer: {
    term: termReducer,
  },
});
