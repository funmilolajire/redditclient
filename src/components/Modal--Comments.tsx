import { ModalComment } from "./Modal--Comment";

interface Comment {
    [key: string]: any
}
interface ModalCommentsType {
    comments: Comment[]
}

export const ModalComments = (props: ModalCommentsType) => {
    const { comments } = props;
    return (
        <div className="Modal--Comments">
            {
                comments.length !== 0 ?
                    <>
                        <h3>Comments</h3>
                        <div className="comments">
                            {comments && comments.map((comment: Comment) => <ModalComment key={comment.id} comment={comment} />)}
                        </div>
                    </> :
                    <p className="no--comments">No comments yet</p>
            }
        </div>
    )
}