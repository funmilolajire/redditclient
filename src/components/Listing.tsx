import { useRef, useLayoutEffect, useCallback } from 'react';
import { useAppDispatch, useAppSelector } from '../store/hooks';
import { handleListingChange, handleTermChange } from '../store/slices/postsSlice';

export const Listing = ({ listItem }: { listItem: string }) => {
    const listingRef = useRef<HTMLLIElement>(null);

    //handle listing change and active class
    const dispatch = useAppDispatch();
    const listing = useAppSelector(state => state.posts.listing);
    const handleListing = () => {
        if (null !== listingRef.current && listing !== listingRef.current.id) {
            let newListing = listingRef.current.id;
            dispatch(handleListingChange(newListing))
            dispatch(handleTermChange(''))
        }
    }
    const handleActiveClass = useCallback(() => {
        if (null !== listingRef.current) {
            const classList = listing === listingRef.current.id ? "Listing active" : "Listing";
            listingRef.current.className = classList
        }
    }, [listing])
    useLayoutEffect(() => {
        handleActiveClass();
    }, [handleActiveClass])


    //JSX
    return (
        <li ref={listingRef} data-testid={listItem} onClick={handleListing} id={listItem}>
            {listItem}
        </li>
    )
}