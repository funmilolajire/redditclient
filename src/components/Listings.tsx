import { useRef, useLayoutEffect } from 'react';
import { Listing } from './Listing';
import { gsap } from 'gsap';

export const Listings = () => {
    const listings: string[] = ['hot', 'new', 'top', 'best', 'rising', 'controversial'];

    //animations
    const listingBoxRef = useRef<HTMLDivElement>(null);
    useLayoutEffect(() => {
        const tl = gsap.timeline({})
        tl.from([listingBoxRef.current], {
            x: '1200%',
            delay: 2,
            duration: 2,
            ease: 'power3.inOut'
        })
    }, [])

    return (
        <div ref={listingBoxRef} className="Listings">
            <ul className="Listings--List">
                {
                    listings && listings.map(listing => <Listing key={listing} listItem={listing} />)
                }
            </ul>
        </div>
    )
}