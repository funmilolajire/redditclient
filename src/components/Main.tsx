import { Listings } from './Listings';
import { Posts } from './Posts';
export const Main = () => (
    <div className="Main">
        <Listings />
        <Posts />
    </div>
)