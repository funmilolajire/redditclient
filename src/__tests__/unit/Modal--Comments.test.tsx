import { mount } from 'enzyme';
import { ModalComments } from '../../components/Modal--Comments';
import { Provider } from 'react-redux';
import { store } from '../../store/store';

describe('renders ModalComments component', () => {
    const comment1 = {
        permalink: '/r/Showerthoughts/comments/mdx3ge/our_buttholes_are_watertight_we_can_go_swimming/gscm02h/',
        time: 1616827642.0,
        author: 'foul_dwimmerlaik',
        comment: 'Gotta protect against ass-eating amoebas',
        upvotes: 10,
        downvotes: 10,
        num_replies: 10,
        id: 1
    }
    const comment2 = {
        permalink: '/r/Showerthoughts/comments/mdx3ge/our_buttholes_are_watertight_we_can_go_swimming/gscm02h/',
        time: 1616827642.0,
        author: 'foul_dwimmerlaik',
        comment: 'Gotta protect against ass-eating amoebas',
        upvotes: 10,
        downvotes: 10,
        num_replies: 10,
        id: 2
    }
    let props = {
        comments: [comment1, comment2]
    }
    let ModalCommentsComponent = mount(
        <Provider store={store}>
            <ModalComments {...props} />
        </Provider>
    )

    test('make a snapshot of ModalComments Component', () => {
        expect(ModalCommentsComponent).toMatchSnapshot()
    })
    test('displays no comments when comments length is 0', () => {
        props = {
            comments: []
        }
        ModalCommentsComponent = mount(
            <Provider store={store}>
                <ModalComments {...props} />
            </Provider>
        )
        const noComments = ModalCommentsComponent.find('.no--comments').text()
        expect(noComments).toBe('No comments yet')
    })
})