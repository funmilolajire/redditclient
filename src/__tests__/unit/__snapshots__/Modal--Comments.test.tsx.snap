// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ModalComments component make a snapshot of ModalComments Component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ModalComments
    comments={
      Array [
        Object {
          "author": "foul_dwimmerlaik",
          "comment": "Gotta protect against ass-eating amoebas",
          "downvotes": 10,
          "id": 1,
          "num_replies": 10,
          "permalink": "/r/Showerthoughts/comments/mdx3ge/our_buttholes_are_watertight_we_can_go_swimming/gscm02h/",
          "time": 1616827642,
          "upvotes": 10,
        },
        Object {
          "author": "foul_dwimmerlaik",
          "comment": "Gotta protect against ass-eating amoebas",
          "downvotes": 10,
          "id": 2,
          "num_replies": 10,
          "permalink": "/r/Showerthoughts/comments/mdx3ge/our_buttholes_are_watertight_we_can_go_swimming/gscm02h/",
          "time": 1616827642,
          "upvotes": 10,
        },
      ]
    }
  >
    <div
      className="Modal--Comments"
    >
      <h3>
        Comments
      </h3>
      <div
        className="comments"
      >
        <ModalComment
          comment={
            Object {
              "author": "foul_dwimmerlaik",
              "comment": "Gotta protect against ass-eating amoebas",
              "downvotes": 10,
              "id": 1,
              "num_replies": 10,
              "permalink": "/r/Showerthoughts/comments/mdx3ge/our_buttholes_are_watertight_we_can_go_swimming/gscm02h/",
              "time": 1616827642,
              "upvotes": 10,
            }
          }
          key="1"
        >
          <div
            className="ModalComment"
            data-permalink="/r/Showerthoughts/comments/mdx3ge/our_buttholes_are_watertight_we_can_go_swimming/gscm02h/"
          >
            <p
              className="Comment--Info"
            >
              comment made 
              <span
                className="time"
              >
                1616827642
              </span>
               by 
              <span
                className="author"
              >
                foul_dwimmerlaik
              </span>
            </p>
            <div
              className="Comment--Body"
            >
              Gotta protect against ass-eating amoebas
            </div>
            <div
              className="Comment--Bottom"
            >
              <div
                className="Comment--Replies"
              >
                <BiComment
                  className="reply-icon icon"
                >
                  <IconBase
                    attr={
                      Object {
                        "viewBox": "0 0 24 24",
                      }
                    }
                    className="reply-icon icon"
                  >
                    <svg
                      className="reply-icon icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M20,2H4C2.897,2,2,2.897,2,4v18l5.333-4H20c1.103,0,2-0.897,2-2V4C22,2.897,21.103,2,20,2z M20,16H6.667L4,18V4h16V16z"
                        key="0"
                      />
                    </svg>
                  </IconBase>
                </BiComment>
                <span
                  className="comments value"
                >
                  10
                </span>
                <span
                  className="label"
                >
                  replies
                </span>
              </div>
              <BsDot
                className="dot-icon"
              >
                <IconBase
                  attr={
                    Object {
                      "fill": "currentColor",
                      "viewBox": "0 0 16 16",
                    }
                  }
                  className="dot-icon"
                >
                  <svg
                    className="dot-icon"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M8 9.5a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"
                      fillRule="evenodd"
                      key="0"
                    />
                  </svg>
                </IconBase>
              </BsDot>
              <div
                className="Votes Comment--Upvotes"
              >
                <GoArrowUp
                  className="arrow-icon icon"
                >
                  <IconBase
                    attr={
                      Object {
                        "viewBox": "0 0 10 16",
                      }
                    }
                    className="arrow-icon icon"
                  >
                    <svg
                      className="arrow-icon icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 10 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M5 3L0 9h3v4h4V9h3L5 3z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </IconBase>
                </GoArrowUp>
                <span
                  className="upvotes value"
                >
                  10
                </span>
                <span
                  className="label"
                >
                  upvotes
                </span>
              </div>
              <BsDot
                className="dot-icon"
              >
                <IconBase
                  attr={
                    Object {
                      "fill": "currentColor",
                      "viewBox": "0 0 16 16",
                    }
                  }
                  className="dot-icon"
                >
                  <svg
                    className="dot-icon"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M8 9.5a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"
                      fillRule="evenodd"
                      key="0"
                    />
                  </svg>
                </IconBase>
              </BsDot>
              <div
                className="Votes Comment--Downvotes"
              >
                <GoArrowDown
                  className="arrow-icon icon"
                >
                  <IconBase
                    attr={
                      Object {
                        "viewBox": "0 0 10 16",
                      }
                    }
                    className="arrow-icon icon"
                  >
                    <svg
                      className="arrow-icon icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 10 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M7 7V3H3v4H0l5 6 5-6H7z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </IconBase>
                </GoArrowDown>
                <span
                  className="downvotes value"
                >
                  10
                </span>
                <span
                  className="label"
                >
                  downvotes
                </span>
              </div>
            </div>
          </div>
        </ModalComment>
        <ModalComment
          comment={
            Object {
              "author": "foul_dwimmerlaik",
              "comment": "Gotta protect against ass-eating amoebas",
              "downvotes": 10,
              "id": 2,
              "num_replies": 10,
              "permalink": "/r/Showerthoughts/comments/mdx3ge/our_buttholes_are_watertight_we_can_go_swimming/gscm02h/",
              "time": 1616827642,
              "upvotes": 10,
            }
          }
          key="2"
        >
          <div
            className="ModalComment"
            data-permalink="/r/Showerthoughts/comments/mdx3ge/our_buttholes_are_watertight_we_can_go_swimming/gscm02h/"
          >
            <p
              className="Comment--Info"
            >
              comment made 
              <span
                className="time"
              >
                1616827642
              </span>
               by 
              <span
                className="author"
              >
                foul_dwimmerlaik
              </span>
            </p>
            <div
              className="Comment--Body"
            >
              Gotta protect against ass-eating amoebas
            </div>
            <div
              className="Comment--Bottom"
            >
              <div
                className="Comment--Replies"
              >
                <BiComment
                  className="reply-icon icon"
                >
                  <IconBase
                    attr={
                      Object {
                        "viewBox": "0 0 24 24",
                      }
                    }
                    className="reply-icon icon"
                  >
                    <svg
                      className="reply-icon icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M20,2H4C2.897,2,2,2.897,2,4v18l5.333-4H20c1.103,0,2-0.897,2-2V4C22,2.897,21.103,2,20,2z M20,16H6.667L4,18V4h16V16z"
                        key="0"
                      />
                    </svg>
                  </IconBase>
                </BiComment>
                <span
                  className="comments value"
                >
                  10
                </span>
                <span
                  className="label"
                >
                  replies
                </span>
              </div>
              <BsDot
                className="dot-icon"
              >
                <IconBase
                  attr={
                    Object {
                      "fill": "currentColor",
                      "viewBox": "0 0 16 16",
                    }
                  }
                  className="dot-icon"
                >
                  <svg
                    className="dot-icon"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M8 9.5a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"
                      fillRule="evenodd"
                      key="0"
                    />
                  </svg>
                </IconBase>
              </BsDot>
              <div
                className="Votes Comment--Upvotes"
              >
                <GoArrowUp
                  className="arrow-icon icon"
                >
                  <IconBase
                    attr={
                      Object {
                        "viewBox": "0 0 10 16",
                      }
                    }
                    className="arrow-icon icon"
                  >
                    <svg
                      className="arrow-icon icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 10 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M5 3L0 9h3v4h4V9h3L5 3z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </IconBase>
                </GoArrowUp>
                <span
                  className="upvotes value"
                >
                  10
                </span>
                <span
                  className="label"
                >
                  upvotes
                </span>
              </div>
              <BsDot
                className="dot-icon"
              >
                <IconBase
                  attr={
                    Object {
                      "fill": "currentColor",
                      "viewBox": "0 0 16 16",
                    }
                  }
                  className="dot-icon"
                >
                  <svg
                    className="dot-icon"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M8 9.5a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"
                      fillRule="evenodd"
                      key="0"
                    />
                  </svg>
                </IconBase>
              </BsDot>
              <div
                className="Votes Comment--Downvotes"
              >
                <GoArrowDown
                  className="arrow-icon icon"
                >
                  <IconBase
                    attr={
                      Object {
                        "viewBox": "0 0 10 16",
                      }
                    }
                    className="arrow-icon icon"
                  >
                    <svg
                      className="arrow-icon icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 10 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M7 7V3H3v4H0l5 6 5-6H7z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </IconBase>
                </GoArrowDown>
                <span
                  className="downvotes value"
                >
                  10
                </span>
                <span
                  className="label"
                >
                  downvotes
                </span>
              </div>
            </div>
          </div>
        </ModalComment>
      </div>
    </div>
  </ModalComments>
</Provider>
`;
